### ISSUE TITLE
Model Reproduction with Full Dataset Encounters Instability - Unable to Reproduce Results

### ISSUE THREAD

**Issue by baijinming97** [myzhengSIMM/TranSiGen#22]
Issue Description
I am experiencing training instability when reproducing TranSiGen on the full dataset. Training proceeds normally for approximately 30 epochs, then becomes unstable. Has anyone successfully reproduced the full training?
Environment Information

Environment: Configured according to official requirements.txt
Data & Code: Downloaded from official GitHub repository
Modification: Only fixed x2 pretrained weight loading path (suspected minor bug in source code)

Problem Details
Training Configuration:

Default hyperparameters: beta=0.1, learning_rate=1e-3
Full LINCS2020 dataset
Official training command as documented

Issue:
Training proceeds normally for ~30 epochs, then becomes unstable with erratic loss behavior and potential convergence issues.

<img width="879" height="591" alt="Image" src="https://github.com/user-attachments/assets/64631e06-bbec-4dad-8f5f-6fc21328105c" />

Request
If you have successfully trained TranSiGen on full data, could you please share:

Confirmation of successful full training completion
Working hyperparameter settings
Any training stability tips for longer training runs

**Comment by baijinming97** [comment]
Hi,

I had KL divergence explosion issues when training with the specified PyTorch 1.5.1.

Updating to PyTorch 2.3.0 fixed it. Training is now stable and PCC matches the paper.

Environment:
- GPU: NVIDIA L40S
- System: Slurm cluster
- PyTorch: 2.3.0 (instead of 1.5.1)

Hope this helps others with similar issues.
